### PR TITLE
add 'Implementing Undo History' recipe

### PR DESCRIPTION
**Pull request by omnidan**
as suggested in https://github.com/rackt/redux/pull/777#issuecomment-142931596 - here is a recipe page about creating the `todos-with-undo` example

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -25,8 +25,9 @@
   * [Migrating to Redux](/docs/recipes/MigratingToRedux.md)
   * [Reducing Boilerplate](/docs/recipes/ReducingBoilerplate.md)
   * [Server Rendering](/docs/recipes/ServerRendering.md)
-  * [Computing Derived Data](/docs/recipes/ComputingDerivedData.md)
```
Comment (omnidan) [inline docs/README.md]: I moved this below `Writing Tests` so it has the same order as in `docs/recipes/README.md`

File: docs/recipes/ImplementingUndoHistory.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+# Implementing Undo History
+
+[redux-undo](https://github.com/omnidan/redux-undo) is a library that provides simple undo/redo functionality for any part of your redux tree.
+
+In this recipe, you will learn how to make the [Todo List example](http://rackt.github.io/redux/docs/basics/ExampleTodoList.html) undoable - all it takes are [two lines of code](https://twitter.com/dan_abramov/status/647040825826918400) and a few terminal commands!
+
+You can find the full source of this recipe in [the redux examples folder](https://github.com/rackt/redux/tree/master/examples/todos-with-undo).
+
+
+## Installing redux-undo
```
Comment (gaearon) [inline docs/recipes/ImplementingUndoHistory.md]: I think that, just before this section, we need a section *explaining* how `redux-undo` works, without mentioning it by name. It doesn't need to be long—we just need to introduce the idea of a higher-order reducer, and how one would go about implementing a trivial `undoable` themselves.

Comment (omnidan) [inline docs/recipes/ImplementingUndoHistory.md]: That's a really good idea! I'll do that :+1:

Comment (omnidan) [inline docs/recipes/ImplementingUndoHistory.md]: @gaearon I did that! :smile:

File: docs/recipes/ImplementingUndoHistory.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
+# Implementing Undo History
+
+To add undo functionality (or any other functionality) to your existing
+reducers, you need to create a higher-order reducer, which is a function
```
Comment (gaearon) [inline docs/recipes/ImplementingUndoHistory.md]: I think this is jumping into the topic too fast. Before the reader understands why they need a reducer => reducer, they need to understand:

* we can express undo history with a regular Redux state, but of a specific form
* we can express undo/redo as regular actions
* we can call write `undoableTodos` reducer that calls `todos` manually
* finally, let's abstract `undoable` out of it!

I understand if this is bigger than you currently have time for—I can revisit this later if you don't find the time.

Comment (omnidan) [inline docs/recipes/ImplementingUndoHistory.md]: It's alright :) I don't mind writing more - I'll try to work on it tomorrow, but I can't promise anything, right now are very busy times for me. Don't worry, I have it noted and will work on it as soon as I find time for it! :grin:

Comment (gaearon) [inline docs/recipes/ImplementingUndoHistory.md]: Yeah no problem. Let's leave it hanging here for now; if you realize you don't have time, it's OK too—just let me know here in the comments so I can finish it.

**Comment by omnidan** [comment]
@gaearon I hope it's good now, I tried to explain everything that redux-undo does :grin:

**Comment by gaearon** [comment]
That's good. :+1: 
Thank you.

I'll go over it when I get some free time, make some style tweaks and merge.

**Comment by omnidan** [comment]
@gaearon Alright, thanks a lot - just take your time :ok_hand:

**Comment by omnidan** [comment]
@gaearon just wanted to give you a heads up that this is still open (in case you forgot) - if you're busy that's okay, I don't want to rush you :grin:

**Comment by gaearon** [comment]
Yeah I'm on vacation now + looking into some other time-demanding stuff, so please give it some time. :-)

**Comment by gaearon** [comment]
Done.

Thank you for your work on this!
I took the liberty of adding some more implementation information because I aimed to explain the pattern too.

**Comment by omnidan** [comment]
@gaearon Alright, nice :grin: Thanks for everything!

**Pull request merged**